client/src/App.jsx

```diff
@@ -4,23 +4,75 @@ import Calendar from './Calendar';
 import Guests from './Guests';
 import Reserve from './Reserve';
 
-
 class App extends React.Component {
   constructor(props) {
     super(props);
+
+    this.state = {
+      guestDropdownVisibility: "hidden",
+      numAdults: 1,
+      numChildren: 0,
+      numInfants: 0,
+      maxGuests: 8, //TODO: update to dynamic from DB
+      disableAdultPlus: false,
+      disableAdultMinus: true,
+      disableChildrenPlus: false,
+      disableChildrenMinus: true,
+      disableInfantPlus: false,
+      disableInfantMinus: true,
+      pricePerNight: 130,
+      starRating: 4.16,
+      numReviews: 10
+    };
+
+    this.toggleGuestsDropdown = this.toggleGuestsDropdown.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.closeGuestsDropdown = this.closeGuestsDropdown.bind(this);
+    this.incrementGuestsCounter = this.incrementGuestsCounter.bind(this);
+    this.decrementGuestsCounter = this.decrementGuestsCounter.bind(this);
+    this.updateAdultChildrenButtonStatus = this.updateAdultChildrenButtonStatus.bind(this);
+    this.updateInfantButtonStatus = this.updateInfantButtonStatus.bind(this);
+    this.updateButtonStatus = this.updateButtonStatus.bind(this);
+    this.displayMaxGuests = this.displayMaxGuests.bind(this);
+    this.displayGuests = this.displayGuests.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('click', this.handleClick, false);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.handleClick, false);
   }
 
   render() {
     return(
```

Comment:
Feel free to refactor based on the comments in later commits.

You can destructure `this.state`'s keys within render before you return.